packages/node/test/unit/install.spec.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  AppABIEncodings,
+  AppState,
+  AssetType,
+  BlockchainAsset
+} from "@counterfactual/types";
+import dotenv from "dotenv-extended";
+import { Wallet } from "ethers";
+import { AddressZero, One, Zero } from "ethers/constants";
+import { instance, mock, when } from "ts-mockito";
+import { v4 as generateUUID } from "uuid";
+
+import { ERRORS, install } from "../../src/methods/install-operations";
+import { openStateChannel } from "../../src/methods/multisig-operations";
+import { ProposedAppInstanceInfo } from "../../src/models";
+import { Store, STORE_ERRORS } from "../../src/store";
+import { EMPTY_NETWORK } from "../integration/utils";
+import MEMORY_STORE_SERVICE from "../services/memory-store-service";
+
+dotenv.load();
+
+describe("Can handle correct & incorrect installs", () => {
+  const storeKeyPrefix = "store";
+
+  it("fails to install without appInstanceId", () => {
+    const store = new Store(MEMORY_STORE_SERVICE, storeKeyPrefix);
+    const params = { appInstanceId: "" };
+    expect(install(store, params)).rejects.toEqual(ERRORS.NO_APP_INSTANCE_ID);
+  });
```

Comment:
Perhaps we should have two tests -- one for invalid appInstanceId (not bytes32, empty string, etc) and one for empty (undefined or null)

Reply:
Yeah that's a fair point. This only covers an empty string because the type checker enforced 
 that on the argument being passed in, but this behavior can't be enforced when this is transpiled to JS so I'll add a second case and override the type checker.